src/state/swap/hooks.ts

```diff
@@ -160,15 +161,25 @@ export function useDerivedSwapInfo(toggledVersion: Version): {
   const bestV3TradeExactIn = useV3TradeExactIn(isExactIn ? parsedAmount : undefined, outputCurrency ?? undefined)
   const bestV3TradeExactOut = useV3TradeExactOut(inputCurrency ?? undefined, !isExactIn ? parsedAmount : undefined)
 
-  const v2Trade = isExactIn ? bestV2TradeExactIn : bestV2TradeExactOut
-  const v3Trade = isExactIn ? bestV3TradeExactIn : bestV3TradeExactOut
+  const v2TradeRaw = isExactIn ? bestV2TradeExactIn : bestV2TradeExactOut
+  const v3TradeState = isExactIn ? bestV3TradeExactIn : bestV3TradeExactOut
+  const v2Trade = v2TradeRaw ?? undefined
+  const v3Trade = v3TradeState.trade && v3TradeState.state == V3TradeState.VALID ? v3TradeState.trade : undefined
```

Comment:
@willpote this isn't ideal since the interfaces uses this trade object to render the ui